src/metabase/query_processor/middleware/results_metadata.clj

```diff
@@ -26,15 +26,16 @@
 ;;
 ;; 2. Consider whether the actual save operation should be async as with
 ;;    [[metabase.query-processor.middleware.process-userland-query]]
-(defn- record-metadata! [{{:keys [card-id]} :info, {:keys [source-card-id]} :query} metadata]
+(defn- record-metadata! [{{:keys [card-id]} :info, :as query} metadata]
   (try
     ;; At the very least we can skip the Extra DB call to update this Card's metadata results
     ;; if its DB doesn't support nested queries in the first place
     (when (and metadata
                driver/*driver*
                (driver/database-supports? driver/*driver* :nested-queries (lib.metadata/database (qp.store/metadata-provider)))
                card-id
-               (not source-card-id))
+               ;; don't want to update metadata when we use a Card as a source Card.
+               (not (:qp/source-card-id query)))
```

Comment:
I tried to change the unnamespaced mystery keys added by the `fetch-source-query` middleware to ones prefixed by `:qp`. This makes it a little more obvious that they came from the QP itself and should make it much easier to grep the codebase for them